app/models/user.rb

```diff
@@ -16,24 +18,23 @@ class User < ActiveRecord::Base
   scope :geocoded,      -> { where(['users.lat != ""']).recent                        }
   scope :alpha,         -> { order('display_name ASC')                                }
 
-  # The before destroy has to be declared *before* has_manys
-  # This ensures User#efficiently_destroy_relations executes first
   before_create :make_first_user_admin
-  before_destroy :efficiently_destroy_relations
+  before_destroy :enqueue_real_destroy_job
```

Comment:
@sudara How about something like this to handle our `cleanup after 30 days` requirement? 

I can def do a sidekiq scheduled job but I feel like we don't delete so many users to have it run every day. Thoughts?

Reply:
This looks great. Users are definitely deleted daily.... we currently have quite the spam problem, but we are talking 10-20 at most!